### PR TITLE
[flink-core] Unify CommitLog and BatchLog source function.

### PR DESCRIPTION
**Pull request by dmvk**

### CODE REVIEW

File: flink/core/src/test/java/cz/o2/proxima/flink/core/BatchLogSourceFunctionTest.java [hunk 1]
```diff
@@ -332,7 +332,12 @@ public void collectWithTimestamp(StreamElement element, long timestamp) {
 
           @Override
           public void go() throws Exception {
-            sourceFunction.run(context);
+            try {
+              sourceFunction.run(context);
+            } catch (Exception e) {
+              e.printStackTrace();
```
Comment (je-ik) [inline flink/core/src/test/java/cz/o2/proxima/flink/core/BatchLogSourceFunctionTest.java:338]: debug?

Comment (dmvk) [reply flink/core/src/test/java/cz/o2/proxima/flink/core/BatchLogSourceFunctionTest.java:338]: yes, I'll try to unify the test cases as well, I think it should be pretty straightforward now

**Comment by dmvk** [comment]
This is ready for merge. I'll leave the test unification for another PR as it seems more complex than I initially thought.

**Comment by dmvk** [comment]
This should also fix the code duplication quality gate.

**Pull request merged**